components-core/src/main/java/com/adyen/checkout/components/core/internal/data/api/AnalyticsMapper.kt

```diff
@@ -75,5 +75,24 @@ class AnalyticsMapper {
         private const val DROP_IN_COMPONENT = "dropin"
         private const val ANDROID_PLATFORM = "android"
         private const val ANDROID_CHANNEL = "android"
+
+        private var VERSION = BuildConfig.CHECKOUT_VERSION
+        private var PLATFORM = ANDROID_PLATFORM
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun overrideVersion(version: String) {
+            VERSION = version
+        }
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun overridePlatform(platform: String) {
+            PLATFORM = platform
+        }
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun resetToDefaults() {
```

Comment:
Could this be needed anywhere other than for testing?

Reply:
Can't think of a use case, but I thought it could be nice to have. Shall I make it `internal`, so only the tests can use it?

Reply:
Yeah let's do this for now, we'll make it public if we need it later